include/nvexec/stream/reduce.cuh

```diff
@@ -110,20 +128,117 @@ namespace nvexec {
+          return __sender_apply((_Sender&&) __sndr, __connect_fn<_Receiver>{__rcvr});
+        }
+      };
+
+      template <class _CvrefSenderId, class _ReceiverId, class _InitT, class _Fun>
+      struct __operation {
+        using _CvrefSender = stdexec::__cvref_t<_CvrefSenderId>;
+        using _Receiver = stdexec::__t<_ReceiverId>;
+        using __receiver_id = receiver_t<_CvrefSender, _ReceiverId, _InitT, _Fun>;
+        using __receiver_t = stdexec::__t<__receiver_id>;
+        _CvrefSender sender;
+        _Receiver rcvr;
+
+        struct __t : __immovable {
+          using __id = __operation;
+          using __data_t = __data<_InitT, _Fun>;
+
+          typename __receiver_id::__op_state __state_;
+          connect_result_t<_CvrefSender, __receiver_t> __op_;
+
+          __t(_CvrefSender&& __sndr, _Receiver __rcvr, __data_t __data) //
+            noexcept(__nothrow_decay_copyable<_Receiver>                //
+                       && __nothrow_decay_copyable<__data_t>            //
+                         && __nothrow_connectable<_CvrefSender, __receiver_t>)
+            : __state_{(__data_t&&) __data, (_Receiver&&) __rcvr}
+            , __op_(connect((_CvrefSender&&) __sndr, __receiver_t{&__state_})) {
+          }
+
+          friend void tag_invoke(start_t, __t& __self) noexcept {
+            start(__self.__op_);
+          }
+
+          template <__decays_to<__t> _Self, receiver _Receivr>
+            requires sender_to<_CvrefSender&, __receiver_t >
+          friend __operation tag_invoke(connect_t, __t& __self, _Receivr __rcvr) noexcept {
+            return {((_Self&&) __self).sender, (_Receivr&&) __rcvr};
+          }
+
+          template <__decays_to<__t> _Self, class _Receivr>
+          friend __operation tag_invoke(connect_t, _Self&& __self, _Receivr __rcvr) noexcept {
+            return {((_Self&&) __self).sender, (_Receivr&&) __rcvr};
+          }
+        };
+      };
+
+      template <class _Receiver>
+      struct __connect_fn {
+        _Receiver& __rcvr_;
+
+        template <class _Child, class _Data>
+        using __operation_t = //
+          __t<__operation<
+            __cvref_id<_Child>,
+            __id<_Receiver>,
+            decltype(_Data::__initT_),
+            decltype(_Data::__fun_)>>;
+
+        template <class _Data, class _Child>
+        auto operator()(__ignore, _Data __data, _Child&& __child) const noexcept(
+          __nothrow_constructible_from<__operation_t<_Child, _Data>, _Child, _Receiver, _Data>)
+          -> __operation_t<_Child, _Data> {
+          return __operation_t<_Child, _Data>{
+            (_Child&&) __child, (_Receiver&&) __rcvr_, (_Data&&) __data};
+        }
       };
 
       template <class SenderId, class InitT, class Fun>
       struct sender_t
         : public __algo_range_init_fun::
             sender_t<SenderId, InitT, Fun, sender_t<SenderId, InitT, Fun>> {
+        using Sender = SenderId;
```

Comment:
This was using `stdexec::__t<SenderId>` before iirc. I changed it because I think i ran into compilation issues. Useless alias.